src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -1100,21 +1114,100 @@ subroutine decomp_rates_mimics(bounds, num_soilc, filter_soilc, &
+            ! https://bg.copernicus.org/articles/9/565/2012/bg-9-565-2012.pdf
+            ! Not moving to the params files at this time: Hardwiring will
+            ! become unnecessary when we calc. ligninNratioAvg the same way for
+            ! FATES and non FATES cases.
+            select case (pftname(patch%itype(p)))
+               case ('not_vegetated')
+                  ligninNratio(p) = 0._r8
+               case ('broadleaf_evergreen_tropical_tree')
+                  ligninNratio(p) = 18.7_r8  ! 17.8 / 0.95
+               case ('broadleaf_deciduous_tropical_tree')
+                  ligninNratio(p) = 8.9_r8  ! 14.5 / 1.63
+               case ('needleleaf_evergreen_temperate_tree')
+                  ligninNratio(p) = 33.4_r8  ! 24.4 / 0.73
+               case ('broadleaf_evergreen_temperate_tree', &
+                     'broadleaf_evergreen_shrub')
+                  ligninNratio(p) = 24.4_r8  ! 21.0 / 0.86
+               case ('broadleaf_deciduous_temperate_tree', &
+                     'broadleaf_deciduous_temperate_shrub')
+                  ligninNratio(p) = 16.6_r8  ! 16.9 / 1.02
+               case ('needleleaf_evergreen_boreal_tree', &
+                     'needleleaf_deciduous_boreal_tree')
+                  ligninNratio(p) = 26.7_r8  ! 25.6 / 0.96
+               case ('broadleaf_deciduous_boreal_tree', &
+                     'broadleaf_deciduous_boreal_shrub')
+                  ligninNratio(p) = 25.1_r8  ! 22.3 / 0.89
+               case ('c4_grass', 'miscanthus', 'irrigated_miscanthus', &
+                     'switchgrass', 'irrigated_switchgrass', &
+                     'tropical_corn', 'irrigated_tropical_corn', &
+                     'temperate_corn', 'irrigated_temperate_corn', &
+                     'sugarcane', 'irrigated_sugarcane', &
+                     'sorghum', 'irrigated_sorghum', &
+                     'millet', 'irrigated_millet')  ! all c4
+                  ligninNratio(p) = 24.1_r8  ! 23.4 / 0.97
+               case default  ! all pfts/cfts that do not fit above categories
+                  ligninNratio(p) = 12.1_r8  ! 16.6 / 1.37
+            end select
+
+         end do  ! p loop
+
+         ! Calculate the column-level averages
+         call p2c(bounds, num_soilc, filter_soilc, &
+              ligninNratio(bounds%begp:bounds%endp), &
+              ligninNratioAvg_local(bounds%begc:bounds%endc))
+         call p2c(bounds, num_soilc, filter_soilc, &
+              annsum_npp(bounds%begp:bounds%endp), &
+              annsum_npp_col_local(bounds%begc:bounds%endc))
+      end if  ! use_fates
+
       ! calculate rates for all litter and som pools
       do fc = 1,num_soilc
          c = filter_soilc(fc)
 
+         if (use_fates) then
+            ligninNratioAvg_scalar = ligninNratioAvg_local(c)
+            annsum_npp_col_scalar = max(0._r8, annsum_npp_col_local(c))
+         else
+            ligninNratioAvg_scalar = cnveg_carbonflux_inst%ligninNratioAvg_col(c)
```

Comment:
I was tracking down how mimics w/o fates arrives at this column level cnveg_carbonflux_inst%ligninNratioAvg_col(c)

Which took me here:

https://github.com/ESCOMP/CTSM/blob/master/src/biogeochem/CNVegCarbonFluxType.F90#L4498-L4503

It looks like its taking an average of the leaf, fineroot and cwd pools.  But I was expecting that the terms ligninNratio_leaf_col(c), ligninNratio_froot_col(c) and ligninNratio_cwd  would all be multiplied by their respective carbon weighting factors (ie leafc_to_litter_col, etc) before being divided through by the sum of those factors.  I probably missed something though, maybe they already are, or I don't understand the units.

Reply:
I think I understand it now, looking here at the leaf term:
https://github.com/ESCOMP/CTSM/blob/master/src/biogeochem/CNVegCarbonFluxType.F90#L4422-L4424
I see that it is pre-multiplied by leafc_to_litter_patch(p). 
Alright, I see no issue.

Reply:
However, I really think there should be units for the definition of the column level average, it wasn't clear to me exactly what this ratio is.  It is defined here:

https://github.com/ESCOMP/CTSM/blob/master/src/biogeochem/CNVegCarbonFluxType.F90#L338

And I think it should have the same units as the multiple of these two factors (using leaf as a guide):

real(r8), allocatable :: lf_flig       (:)   ! leaf litter lignin fraction
real(r8), allocatable :: lflitcn       (:)   ! leaf litter C:N (gC/gN)

So it seems like this column level average is the mean C:N ratio of all lignin tissues in the column.  [gC/gN]

Reply:
Thanks for diving in here @rgknox.  @slevisconsulting maybe some additional comments in the code are warranted to clarify what's going on?